pandas/core/dtypes/cast.py

```diff
@@ -239,7 +236,9 @@ def trans(x):
                 if (new_result == result).all():
                     return new_result
             else:
-                if np.allclose(new_result, result, rtol=0):
+                # np.allclose raises TypeError on extension arrays
```

Comment:
Hmm not sure about this. I think allclose *should* still work and maybe that is the root cause instead, but let's see what others think

Reply:
i agree , we might need to enable soething on EAs to make this work, but that is the root cause

Reply:
Can you elaborate on enabling something on EAs? Would this be a new feature to it, or a current feature?

Reply:
what happens when this is on an EA (e.g. show an exampel with this line specifically)